packages/magento-product/components/AddProductsToCart/AddProductsToCartForm.tsx

```diff
@@ -80,6 +82,19 @@ export function AddProductsToCartForm(props: AddProductsToCartFormProps) {
             entered_options: cartItem.entered_options?.filter((option) => option?.value),
           })),
       }
+
+      const lastItem = requestData.cartItems[requestData.cartItems.length - 1]
+      const { sku } = lastItem
+
+      if (sku) {
+        // Preload crosssells
+        await client.query({
+          query: CrosssellsDocument,
+          variables: { pageSize: 1, filters: { sku: { eq: sku } } },
+        })
+      }
+
+      return requestData
```

Comment:
We are now waiting for the crossells query to be completed _before_ we are submitting the form. What we want:
1. Fire of the Crosssells query in onBeforeSubmit
2. Wait for the result in of the query in onComplete.

You can probably store the promise in a useRef.

Reply:
And second, we only want to do this `if(redirect="added")`

Reply:
@paales ✅ Fixed